libvcs4j-spoon/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/SpoonModel.java

```diff
@@ -0,0 +1,336 @@
+        }
+    }
+
+    /**
+     * Creates a temporary directory in which the class files get stored. This temporary directory
+     * will be deleted on shutdown (see {@link #recursiveDeleteOnShutdownHook(Path)}).
+     */
+    private void createTmpDir() {
+        try {
+            temporaryDirectory = Files.createTempDirectory("tmpClassDirectory").toFile();
+            recursiveDeleteOnShutdownHook(temporaryDirectory.toPath());
+        } catch (IOException e) {
+            LOGGER.error("Failed creating temporary directory");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * This method adds a shutdown hook to the VM. This hook deletes the given directory and
+     * all its subdirectories/files.
+     *
+     * @param path
+     *      The absolute path to the directory that should be deleted on shutdown.
+     */
+    private void recursiveDeleteOnShutdownHook(final Path path) {
+        Runtime.getRuntime().addShutdownHook(new Thread(
+                () -> {
+                    LOGGER.info("About to delete " + path.toString());
+                    try {
+                        Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+                            @Override
+                            public FileVisitResult visitFile(final Path file, @SuppressWarnings("unused") BasicFileAttributes attrs)
+                                    throws IOException {
+                                Files.delete(file);
+                                return FileVisitResult.CONTINUE;
+                            }
+
+                            @Override
+                            public FileVisitResult postVisitDirectory(final Path dir, final IOException e)
+                                    throws IOException {
+                                if (e == null) {
+                                    Files.delete(dir);
+                                    return FileVisitResult.CONTINUE;
+                                }
+                                // directory iteration failed
+                                throw e;
+                            }
+                        });
+                    } catch (IOException e) {
+                        throw new RuntimeException("Failed to delete " + path, e);
+                    }
+                }));
+    }
+
+    /**
+     * Returns all old or new files from the given list of {@link FileChange} objects. Only java source
+     * files will be contained in the output. The parameter {@code oldFile} indicates if the old or new files
+     * should be extracted.
+     *
+     * @param fileChanges
+     *      The list of {@link FileChange} objects.
+     * @param oldFile
+     *      If true, the old files get extracted, otherwise the new files.
+     * @return
+     *      List of strings containing all paths to the old or new files
+     */
+    private List<String> getAllFilesFromFileChanges(final List<FileChange> fileChanges, final boolean oldFile) {
```

Comment:
I suggest to use an enum rather than a boolean. Enums are way more readable:

```java
getAllFilesFromFileChanges(files, true)
getAllFilesFromFileChanges(files, OLD_FILES)
```